src/js/index.js

```diff
@@ -133,16 +149,87 @@ document.addEventListener("DOMContentLoaded", function () {
   init();
 });
 
-// 1. 브라우져 렌더링 과정
-// dom tree cssom tree render tree
-// script tag (defer async)
 
-// 2. DOMContentLoaded, window.onload 돔이 렌더링이 된 이후에 울리는 함수 (브라우져)
+// delegate 함수 구현 (중간중간 console.log) 고도화
+
+// 유사배열, 배열
+
+// 아래 함수 구현
+
+// 생활코딩 git 수업
+
+// 페이지를 로드
+function loadMenu() {
+  localStorage.setItem('espresso', JSON.stringify({
+      espresso: [{
+          id: 1,
+          name: 'papico'
+      }, {
+          id: 2,
+          name: 'siwon'
+      }]
+  }));
 
-// 3. querySelectorAll, querySelector
+  loging();
+}
+
+loadMenu();
+
+
+// 메뉴추가
+function addMenu() {
+  const espressoMenuList = JSON.parse(localStorage.getItem('espresso'));
+  
+  const newMenu = {
+      id: 3,
+      name: 'cool-jung'
+  };
+  const copyEspressoMenuList = [...espressoMenuList.espresso,newMenu];
+  localStorageSetItem({espresso,copyEspressoMenuList});
+}
+addMenu();
+
+// 메뉴 삭제
+function removeMenu() {
+  const espressoMenuList = JSON.parse(localStorage.getItem('espresso'));
+  console.log(espressoMenuList.espresso);
+  const deleteMenuId = 1;
+  const removeMenuList = espressoMenuList.espresso.filter((element)=>{
+      return element.id !== deleteMenuId;
+  });
+  console.log(removeMenuList);
+  
+  localStorageSetItem({espresso,removeMenuList});
+}
+removeMenu();
+
+// 메뉴 수정
+function editMenu(){
+  const espressoMenuList = JSON.parse(localStorage.getItem('espresso'));
+  const editMenu = {
+      id: 2,
+      name: 'luya'
+  };
+  const updatedName = espressoMenuList.espresso.map((element)=>{
+      if(element.id == 2){
+          element.name = editMenu.name;
+          return element;
+      }else{
+          return element;
+      }
+  });
+  console.log(updatedName);
+  localStorageSetItem({espresso,updatedName});
```

Comment:
```suggestion
  localStorageSetItem({ key: 'espresso', val: updatedName });
```

함수에서 객체를 넘기는 부분은 저런 방식으로 표현해야합니다!

Reply:
익숙해질때까지 연습하시면 좋을 것 같아요!